modules/statistics-provider-matomo/src/main/java/org/opencastproject/statistics/provider/matomo/StatisticsProviderMatomoService.java

```diff
@@ -0,0 +1,175 @@
+
+  /**
+   * Get an existing batch request for a specific ResourceType and Matomo API method
+   * @param resourceTypeMethod The ResourceType concatenated with the Matomo API method
+   * @return The existing batch request or null if none exists
+   */
+  public BatchMatomoRequest getBatchRequest(String resourceTypeMethod) {
+    return methodToBatchRequest.get(resourceTypeMethod);
+  }
+
+  /**
+   * Register a new batch request for a specific ResourceType and Matomo API method
+   * @param resourceTypeMethod The ResourceType concatenated with the Matomo API method
+   * @param batchRequest The batch request to register
+   */
+  public void registerBatchRequest(String resourceTypeMethod, BatchMatomoRequest batchRequest) {
+    methodToBatchRequest.put(resourceTypeMethod, batchRequest);
+  }
+
+  @Reference
+  public void setStatisticsCoordinator(StatisticsCoordinator service) {
+    this.statisticsCoordinator = service;
+  }
+
+  @Activate
+  public void activate(ComponentContext cc) {
+    logger.info("Activating Statistics Provider Matomo Service");
+  }
+
+  @Deactivate
+  public void deactivate(ComponentContext cc) {
+    logger.info("Deactivating Statistics Provider Matomo Service");
+  }
+
+  @Override
+  public void install(File file) throws Exception {
+    final String json = new String(Files.readAllBytes(file.toPath()), Charset.forName("utf-8"));
+    final MatomoProviderConfiguration providerCfg = MatomoProviderConfiguration.fromJson(json);
+    StatisticsProvider provider;
+    switch (providerCfg.getType().toLowerCase()) {
+      case "timeseries": {
+        provider = new MatomoTimeSeriesStatisticsProvider(
+                this,
+                providerCfg.getId(),
+                providerCfg.getResourceType(),
+                providerCfg.getTitle(),
+                providerCfg.getDescription(),
+                providerCfg.getSources());
+        logger.info("Installed Matomo time series statistics provider '{}'", providerCfg.getId());
+      }
+      break;
+      default:
+        throw new ConfigurationException("Unknown Matomo statistics type: " + providerCfg.getType());
+    }
+    fileNameToProvider.put(file.getName(), provider);
+    statisticsCoordinator.addProvider(provider);
+  }
+
+  @Override
+  public void uninstall(File file) {
+    if (fileNameToProvider.containsKey(file.getName())) {
+      statisticsCoordinator.removeProvider(fileNameToProvider.get(file.getName()));
+      fileNameToProvider.remove(file.getName());
+    }
+  }
+
+  @Override
+  public boolean canHandle(File file) {
+    return "statistics".equals(file.getParentFile().getName())
+        && file.getName().endsWith(".json")
+        && file.getName().toUpperCase().startsWith("matomo.".toUpperCase());
+  }
+
+  @Override
+  public void update(File file) throws Exception {
+    uninstall(file);
+    install(file);
+  }
+
+  @Override
+  public void updated(Dictionary<String, ?> dictionary) {
```

Comment:
Caused by removing the ManagedService interface.
```suggestion
  @Modified
  public void modified(Map<String, Object> properties) {
```

Reply:
👍